mitodl/ocw-hugo-themes
# feat: update webpack config; HugoServer and Webpack Port to be same (3000)

#### Pre-Flight checklist


#### What are the relevant tickets?
#1096 

#### What's this PR do?
This PR updates the webpack configuration for development such that the hugoServer and Webpack port is same(port 3000).
Previously hugoServer was at port 3000, and webpack port at 3001 which caused issues related to dynamic importing. It also introduces webpack watch mode for dev.

#### How should this be manually tested?
This PR requires to check that the structure is working as intended, and that all the modules used for building the website work fine.

Steps:
1. Clear your browser cache
2. Checkout to this branch.
3. `yarn fmt`(check that its working)
4. `yarn start course` and `yarn start www`
5. Making sure ocw-www and ocw-course work as it was before.
6. Making sure the webpack related modules/plugins are working i.e the assets, css styling, imports, js files, fonts, images, social media icons on footer, etc
7. Check inspect element and ensure there are no errors related to this.
8. Making sure lecture videos are playing on course.
9. After disconnecting from server, do run`yarn fmt` again.

To test production webpack config:
1. Clear your browser cache
2. Checkout to this branch
3. Open the file: `package.json`
Modify line#10:
From:
`    "start:webpack": "yarn with-env --dev -- webpack --config ./base-theme/assets/webpack/webpack.dev.ts --stats-children",`
To:
`    "start:webpack": "yarn with-env --dev -- webpack --config ./base-theme/assets/webpack/webpack.prod.ts --stats-children",
`
4. Same above steps from 3-9.
